Assets/__Scripts/Singletons/Adb.cs

```diff
@@ -0,0 +1,293 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.Networking;
+
+
+namespace QuestDumper
+{
+    public readonly struct AdbOutput
+    {
+        public readonly string StdOut;
+        public readonly string ErrorOut;
+
+        public AdbOutput(string stdOut, string errorOut)
+        {
+            StdOut = stdOut;
+            ErrorOut = errorOut;
+        }
+
+        public override string ToString()
+        {
+            return $"{nameof(StdOut)}: {StdOut}, {nameof(ErrorOut)}: {ErrorOut}";
+        }
+    }
+
+    /// <summary>
+    /// This code was generously given by Cyuubi#4701
+    /// Much appreciated :)
+    /// </summary>
+    public static class Adb
+    {
+        private const string PLATFORM_TOOLS_DOWNLOAD_GENERIC =
+            "https://dl.google.com/android/repository/platform-tools-latest-";
+
+        private static Process _process;
+
+        private static string GetFullPath(string fileName)
+        {
+            if (File.Exists(fileName))
+                return Path.GetFullPath(fileName);
+
+            fileName = Path.GetFileName(fileName);
+
+            var values = Environment.GetEnvironmentVariable("PATH");
+
+            return values?.Split(Path.PathSeparator).Select(path => Path.Combine(path, fileName)).FirstOrDefault(File.Exists);
+        }
+
+        private static string GetADBUrl()
+        {
+            switch (Application.platform)
+            {
+                case RuntimePlatform.OSXEditor:
+                case RuntimePlatform.OSXPlayer:
+                    return PLATFORM_TOOLS_DOWNLOAD_GENERIC + "darwin.zip";
+                case RuntimePlatform.WindowsPlayer:
+                case RuntimePlatform.WindowsEditor:
+                    return PLATFORM_TOOLS_DOWNLOAD_GENERIC + "windows.zip";
+                case RuntimePlatform.LinuxPlayer:
+                case RuntimePlatform.LinuxEditor:
+                    return PLATFORM_TOOLS_DOWNLOAD_GENERIC + "linux.zip";
+                default:
+                    throw new InvalidOperationException("How could this even happen?");
```

Comment:
Yeah... who would try and run this on android.... oh wait
![Android](https://cdn.discordapp.com/attachments/702231982335197264/857745774308753408/Screenshot_20210624-151520_ChroMapper.jpg)

On a more serious note it may be better to use compiler macros so that this doesn't have to be checked at runtime.

Reply:
That would be lovely, how would I go about doing that with Unity? Also I was funnily enough going to add Android support just for the joke (especially since Quest is Android, mapping in VR ???)

Reply:
https://github.com/Caeden117/ChroMapper/blob/41a64f50212de47b252a7d33881cfab8f78aea32/Assets/StandaloneFileBrowser/StandaloneFileBrowser.cs#L18

Is probably the most complete example although we have a couple other checks just for windows like:

https://github.com/Caeden117/ChroMapper/blob/41a64f50212de47b252a7d33881cfab8f78aea32/Assets/__Scripts/UI/Contributors/ContributorListItem.cs#L90